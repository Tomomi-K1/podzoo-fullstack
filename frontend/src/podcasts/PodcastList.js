import React, { useState, useEffect } from "react";
import PodcastCard from "./PodcastCard";
// Material UI
import Grid from '@mui/material/Grid';

/** PodcastList : stateless
 * receives podcasts as a prop
 * shows list of podcasts using PodcastCard
 * component using PodcastList: Home, SearchResults, FavoriteList
 */
function PodcastList({podcasts}){
    return(
        // add hover effect
        <Grid container spacing={1} sx={{flexGrow:1}} justifyContent="center">
            <Grid item xs={12} md={11} lg={10} sx={{ display:"flex", justifyContent:"center"}}>
                <Grid container justifyContent="center" spacing={1}>
                    {podcasts.map(podcast =>{
                        return (         
                            <Grid key={podcast.feedId} item xs ={5} md={3} lg={2} sx={{display:"flex", justifyContent:"center"}}>
                                <PodcastCard podcast={podcast} />
                            </Grid>
                        
                        )})
                    }
                </Grid>
            </Grid>
        </Grid>
    )
}

export default PodcastList;