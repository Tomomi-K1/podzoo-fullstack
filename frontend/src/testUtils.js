import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };