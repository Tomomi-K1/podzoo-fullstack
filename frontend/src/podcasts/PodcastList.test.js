import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import PodcastList from "./PodcastList";
import { UserProvider } from "../testUtils";

const podcasts =[{
    author:'test-author',
    title:'test-title',
    artwork:'test-artwork',
    feedId: 1
}]

it('renders', () => {
    render(
    <MemoryRouter>
        <UserProvider>
            <PodcastList podcasts={podcasts} />
        </UserProvider>
    </MemoryRouter>
    );
})

it('matches snapshot: user logged in', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <PodcastList podcasts={podcasts} />
            </UserProvider>
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('it has podcasts data', () => {
    render(
    <MemoryRouter>
        <UserProvider>
            <PodcastList podcasts={podcasts} />
        </UserProvider>
    </MemoryRouter>
    );
    expect(screen.getByText('test-title')).toBeInTheDocument();
})