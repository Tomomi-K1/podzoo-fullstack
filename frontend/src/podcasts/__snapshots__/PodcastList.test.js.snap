// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot: user logged in 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-td5piu-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-11 MuiGrid-grid-lg-10 css-x02yn7-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-exfl4s-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-md-3 MuiGrid-grid-lg-2 css-1o44jp4-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1i59dwk-MuiPaper-root-MuiCard-root"
          >
            <a
              href="/podcast/1"
            >
              <img
                alt="podcast artwork"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="auto"
                src="test-artwork"
                width="100%"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1vweose-MuiTypography-root"
                >
                  test-title
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-cq90jg-MuiTypography-root"
                >
                  test-author
                </p>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
